doxylicious/README.md

```diff
@@ -0,0 +1,229 @@
+* optionally, steps to be taken to verify that this procedure worked.
+
+Procedures need to be tested to confirm that the documented steps actually work with (a release of) the software they're describing.
+
+This module enables defining a procedure in a form which can be used both
+* to generate the procedure documentation, in AsciiDoc 
+* to execute the procedure as a test, using the Junit 5 framework
+
+In order words, this makes it possible to automatically test some of your documentation, eliminating toil and the possibility of divergence between what's documented and what actually works.
+
+## Modules 
+
+* `doxylicious-core` defines all the core functionality.
+* `doxylicious-junit-provider` implements a Junit5 extension to enable execution of procedures as tests.
+* `doxylicious-maven-plugin` implements a Maven plugin for generating the procedure AsciiDocs.
+
+## Machine-readable procedures
+
+AsciiDoc is a fine format for authoring documentation, but was never intended to be executable. 
+These tools instead accept procedures defined in YAML (or JSON if you prefer) as a single source of truth for procedures.
+The AsciiDoc form of the procedure can be generated from the YAML.
+
+### A basic proc
+
+Let's start with a simple procedure
+
+```yaml
+id: succeeding_reliably # <1>
+title:
+  adoc: Succeeding reliably # <2>
+abstract:
+  adoc: 'This procedure is a minimal example.' # <3>
+intro:
+  adoc: |- # <4>
+    This procedure will always succeed because the only thing actually executed is the command `true`, which always returns 
+     with a 0 error code.
+procedure:
+  - step: # <5>
+      - adoc: Execute `true`.
+      - exec: true
+      - adoc: It is expect that this command returns without any output.
+  - step:
+      - adoc: Drink some coffee.
+verification:
+    - step: # <6>
+          - adoc: Check that coffee was drunk.
+          - exec: true
+```
+
+1. Each procedure must have a unique `id`.
+2. Published procedures¹ must have an AsciiDoc `title`.
+3. Published procedures may have an AsciiDoc `abstract`.
+4. Published procedures may have an AsciiDoc `intro`.
+5. This procedure has two steps, the first of which has three substeps. A substep:
+    * either specifies some words, using `adoc`,
+    * or specifies a command to be executed, using `exec`.
+6. Optionally, procedures may have `verification` steps.
+
+<small>¹ See the section on generating documentation for what 'published' means.</small>
+
+### A proc with prerequisites
+
+```yaml
+id: defining_prereqs 
+title:
+  adoc: Defining prerequisites
+abstract:
+  adoc: 'This procedure demonstrates prerequisites.' 
+prereqs: # <1>
+  - ref: succeeding_reliably # <2>
+    adoc: You should have already succeeded # <3>
+procedure:
+  - step: # <5>
+      - adoc: Execute `true` again
+      - exec: true
+```
+
+1. A proc provides a list of other procedures that must be done before this one. 
+2. The `ref` is the id of the prerequisite procedure.
```

Comment:
What is the scope of ID resolution? Is it just within the (for want of a better term) the docpath of the current build? Are ID's expected to be globally unique or just within the scope of an execution? 

Could I write a procedure that has a prerequisite defined in a different publication or source tree?

Reply:
>  Is it just within the (for want of a better term) the docpath of the current build?

Yes.

> Are ID's expected to be globally unique or just within the scope of an execution?

Within the scope of an execution. 

> Could I write a procedure that has a prerequisite defined in a different publication or source tree?

In principal? Yes, you could layer on a publishing and distribution model. E.g. if a project published its procedures in YAML format somewhere then a separate project could "import" it and use the procedures it defined (e.g. as prereqs).